sherpa/astro/tests/test_astro_plot.py

```diff
@@ -25,22 +28,172 @@
 
 from sherpa.astro.data import DataPHA
 from sherpa.astro.plot import SourcePlot, \
-    DataPHAPlot, ModelPHAHistogram
+    DataPHAPlot, ModelPHAHistogram, EnergyFluxHistogram, PhotonFluxHistogram
 from sherpa.astro import plot as aplot
 from sherpa.data import Data1D
 from sherpa.models.basic import Const1D, Gauss1D, Polynom1D
 from sherpa import stats
 from sherpa.utils.err import IOErr
-from sherpa.utils.testing import requires_pylab
 
 
-def test_sourceplot():
+def check_sourceplot_energy(sp, rate=True, factor=0):
+    """Check for test_sourceplot/test_sourceplot_channel
+
+    rate determines if the plot is for a "rate" or "counts",
+    although at the moment it does not change anything.
+    """
+
```

Comment:
Rather than stub in rate, the API seems straight forward enough to add rate  in later when you are ready to use it (since the rest of the script appears to be using named rather than positional parameters). The runtime savings are negligible but the code looks cleaner without unused hooks/stubs and comments noting they are unused.